docs: Search Metadata Batch 04 (#14596)
hashicorp/consul: website/content/docs/connect/gateways/mesh-gateway/wan-federation-via-mesh-gateways.mdx

---
layout: docs
page_title: Mesh Gateways for WAN Federation
description: >-

> You can use mesh gateways to federate deployments, which enables WAN gossip between agents in different datacenters. Forward service requests, use prepared queries, and replicate key/value entries across geographically separated clusters by configuring a primary datacenter.

Reviewer: ```suggestion
You can use mesh gateways to federate Consul datacenters. Enable WAN gossip between Consul servers, forward service requests, and use prepared queries across geographically separated clusters by configuring a primary datacenter.
```

You can't replicate k/v.

Reviewer 2: I also thought that prepared queries didn't go through mesh gateways (or that there is some other complication when trying to access prepared query upstreams when a gateway is involved). Maybe we should leave out prepared queries.

```suggestion
You can use mesh gateways to simplify the networking requirements to WAN federate Consul datacenters, including reducing cross-datacenter connection paths, ports, and communication protocols.
```

Also, in the near future, you'll be able to use mesh gateways for multi-DC control plane communication based on cluster peering, not just based on WAN federation. This description is fine though (because the page currently only covers WAN federation).

Reviewer 2: This page also seems to be about control plane communciation (federation), not data plane communication.
